### DISCUSSION TITLE
Jellyfin through VPN

### DISCUSSION THREAD

**Discussion by st0rm-bless3d** [AdrienPoupa/docker-compose-nas#57] [Q&A]
I currently have everything going through VPN, except Jellyfin. When I added the network mode: VPN it gave me:

"Error response from daemon: conflicting options: port publishing and the container type network mode"

I took it off VPN, and everything spun up normally.

I am a little green to some to of this stuff but everything is working besides this, and I know its an optional thing to have Jellyfin on the VPN but it is a personal preference for myself. 

What am I missing in my docker-compose.yml that will make this possible?

Thanks!

**Comment by AdrienPoupa** [comment]
Hi there, I take it you've applied this: https://github.com/AdrienPoupa/docker-compose-nas?tab=readme-ov-file#optional-using-the-vpn-for-arr-apps

If you use network mode service, you can't expose ports in this container anymore. Remove this (auto discovery ports):

```yaml
    ports:
      - "7359:7359/udp"
      - "1900:1900/udp"
 ```

If you want to expose the ports, maybe add them to the vpn container itself?

**Reply by st0rm-bless3d** [reply]
Thanks for getting back to me! Okay maybe a stupid question, is there a reason I want these ports exposed in the first place? Just making sure I am not screwing something else up by doing this.

**Reply by AdrienPoupa** [reply]
They are used for auto discovery, see list here: https://jellyfin.org/docs/general/networking/#port-bindings
